### ISSUE TITLE
requireParametersWithoutDefault does not work anymore

### ISSUE THREAD

**Issue by TempleClause** [RicoSuter/NSwag#2816]
So I read about that with API Explorer query parameters are now optional by default. 
In https://github.com/RicoSuter/NSwag/issues/1556 it was mentioned that setting RequireParametersWithoutDefault to true will keep the old behavior, but this does not seem to work. Is this feature still supported? Is there another way to make parameters required by default? 

Adding the [BindRequired] to a parameter works but we won't to avoid having to add that to all our parameters. 

Related: https://github.com/RicoSuter/NSwag/issues/1599

Thanks

**Comment by RicoSuter** [comment]
I dont remember that there is a recent change there. We need to check the code for this feature. The simplest solution is to implement and register a custom operation processor which just changes/patches all parameters.

**Comment by TempleClause** [comment]
Thank you for your quick answer. 

Can you please point me in the right direction how to implement and register a custom operation processor?

**Comment by TempleClause** [comment]
On further investigation I found out that the `RequireParametersWithoutDefault` works if I use it like this:

```
  services.AddOpenApiDocument(document =>
            {
                document.DocumentName = "v2";
                document.RequireParametersWithoutDefault = true;
            });
```

But it doesn't work if I use it in the nswag config file.

**Comment by RicoSuter** [comment]
Yes probably because the config uses the UseServiceProvider = true (recommended!) which ignores lots of configs and instead uses the config in code (so that the spec at runtime is always the same as in cli)

**Comment by tjmcdonough** [comment]
Is it possible to add a default value to all nullable parameters? I want 
`string? param1` to be `string? param1 = null`
